mailbbgun/views.py

```diff
@@ -0,0 +1,160 @@
+import flask
+import jsonschema
+import logging
+import pika
+
+from mailbbgun import app
+from mailbbgun import models
+
+
+if app.config['DEBUG']:
+    logging.basicConfig(level=logging.DEBUG)
+
+
+_DELIVERY_MODE_PERSISTENT = 2
+_LOG = logging.getLogger(__name__)
+
+
+@app.route('/messages', methods=['POST'])
+def new_message():
+    data = _validate_message_request()
+    _LOG.debug("New message: {}".format(data))
+
+    message = models.Message(**data)
+    models.db.session.add(message)
+    models.db.session.commit()
+
+    connection = pika.BlockingConnection(
```

Comment:
I actually think it would be worse if this process dies here, since it would be harder for us to notice this failure.  If rabbit dies, we would know to go requeue pending messages, but in this case we may never notice.  The client will see a timeout, and then they may or may not attempt to resend depending on whether they notice the pending message when they get the list of messages.

I think it would be better to move the `models.db.session.commit()` call to be the last operation in this view.  If the process dies, or rabbit fails for some reason, the client will still see the error but the session will be rolled back, and no data will be written to the DB.  Also, if for some reason the session commit fails after the message is queued, the worker will get a message that won't be found in the DB.  But the worker will just log the error and continue processing other messages.